Divyanshu | Test | Read bank statement from custom mail and import transactions into Giddh (email forwarding)

dvCodeWorld opened the pull request:
https://app.clickup.com/t/86czty1dn

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

# Release Notes

* **New Features**
  * Introduced email forwarding feature with multi-step onboarding, account management, and verification workflow
  * Reorganized navigation menu with new Automation group consolidating OCR and bank integration options

* **Style**
  * Added new sizing utilities for enhanced layout control (min/max width and height classes)
  * Updated utility classes with improved border and spacing options

* **Documentation**
  * Added multi-language support for email forwarding feature (English, Hindi, Marathi)

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dvCodeWorld committed:
Based on the git diff and following the repository's commit message style, here's the commit message:

```
feat: add bank statement module with onboarding, list, and stepper form components

- Created BankStatementModule with routing configuration for onboarding, list, create, and edit paths
- Added DataListComponent with Material table for displaying bank statements with search, filter, sort, and pagination
- Implemented status badges, transaction info display, and action buttons (view, edit, download, delete)

dvCodeWorld committed:
refactor: update data-list component to display email forwarding configurations instead of bank statements

- Changed component to fetch and display email forwarding data from BankStatementComponentStore
- Updated table columns from bank statement fields (bankName, accountNumber, statementDate, transactions) to email forwarding fields (forwardedMail, bankAccountName, uniqueName, originalEmail)
- Replaced mock data loading with store subscriptions for emailForwardingList$ and isLoading$
- Updated action

dvCodeWorld committed:
refactor: simplify data-list component UI and improve navigation flow

- Removed mat-card wrapper and simplified table layout structure
- Removed sorting, pagination, and multiple action buttons (view, edit, copy)
- Changed table rows to be clickable for editing, kept only delete action button
- Updated editStatement to accept EmailForwardingResponse object and navigate with step query param
- Added subscription for deleteEmailForwardingIsSuccess$ to reload data after deletion
- Added subscription

dvCodeWorld committed:
refactor: remove unused Material modules and add localization support to bank statement data-list

- Removed unused Material modules (Menu, Checkbox, Tabs, Ripple, Chips, Input, Dialog, SlideToggle, List, Scrolling, Sort, Paginator, Select, Datepicker, NativeDate, Divider) from BankStatementModule
- Removed unused helper modules (ClickOutside, DigitsOnly, ElementViewChild, NgxMask, WatchVideo)
- Added TranslateDirectiveModule and NewConfirmationModalModule imports
- Wrapped data-list template with

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/bank-statement.module.ts:
```diff
@@ -0,0 +1,56 @@
+        FormsModule,
+        ReactiveFormsModule,
+        BankStatementRoutingModule,
+        LaddaModule.forRoot({
```
remove module

dvCodeWorld replied:
done

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/bank-statement.module.ts:
```diff
@@ -0,0 +1,56 @@
+            style: 'slide-left',
+            spinnerSize: 30
+        }),
+        SharedModule,
```
please check modules

dvCodeWorld replied:
Done

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/bank-statement.routing.module.ts:
```diff
@@ -0,0 +1,20 @@
+  { path: '', redirectTo: 'onboarding', pathMatch: 'full' },
+  { path: 'onboarding', component: OnboardingComponent },
+  { path: 'list', component: DataListComponent },
+  { path: 'create', component: StepperFormComponent },
```
please correct form name, do according to module wise

dvCodeWorld replied:
Done

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/components/data-list/data-list.component.scss:
```diff
@@ -0,0 +1,150 @@
+//     }
+// }
+
+// .table-container {
```
remove commented css

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/components/data-list/data-list.component.ts:
```diff
@@ -0,0 +1,200 @@
+import { GeneralService } from '../../../services/general.service';
+
+@Component({
+    selector: 'app-data-list',
```
please do data-list

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/components/data-list/data-list.component.ts:
```diff
@@ -0,0 +1,200 @@
+            takeUntil(this.destroyed$)
+        ).subscribe((emailForwardingList) => {
+            if (emailForwardingList) {
+                if (emailForwardingList?.length  === 0) {
```
```suggestion
                if (!emailForwardingList.length) {
```

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/components/data-list/data-list.component.ts:
```diff
@@ -0,0 +1,200 @@
+        ).subscribe((emailForwardingList) => {
+            if (emailForwardingList) {
+                if (emailForwardingList?.length  === 0) {
+                    this.router.navigate(['pages/bank-statement/onboarding']);
```
this.router.navigate(['pages', 'bank-statement', 'onboarding']);

please use this format

dvCodeWorld replied:
make all same pages/bank-statement/*

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/components/data-list/data-list.component.ts:
```diff
@@ -0,0 +1,200 @@
+     * @memberof DataListComponent
+     */
+    public copyEmail(email: string): void {
+        if (navigator.clipboard) {
```
use giddh clipboard service for copy content

dvCodeWorld replied:
done

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/components/data-list/data-list.component.ts:
```diff
@@ -0,0 +1,200 @@
+        this.router.navigate([`/pages/bank-statement/edit/${element.uniqueName}`], { queryParams });
+    }
+
+    
```
remove white space

dvCodeWorld replied:
done

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/components/onboarding/onboarding.component.ts:
```diff
@@ -0,0 +1,99 @@
+import { BankStatementComponentStore } from '../../store/bank-statement.store';
+
+@Component({
+    selector: 'app-onboarding',
```
do for all components

dvCodeWorld replied:
done

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/components/stepper-form/stepper-form.component.html:
```diff
@@ -0,0 +1,212 @@
+                                                </div>
+                                                <p class="mb-3">
+                                                    <a
+                                                        href="https://giddh.com/help/integration/how-to-integrate-email-for-sending-an-invoice"
```
user constant variable here

dvCodeWorld replied:
done

ankitdubeyoffice commented on apps/web-giddh/src/app/email-forwarding/components/create/create.component.html:
```diff
@@ -0,0 +1,212 @@
+                                                    [label]="commonLocaleData?.app_account"
+                                                    [placeholder]="commonLocaleData?.app_select_account"
+                                                    [options]="accountSearchResponse$ | async"
+                                                    formControlName="accountUniqueName"
```
please use showError for error

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/components/stepper-form/stepper-form.component.ts:
```diff
@@ -0,0 +1,494 @@
+import { API_BULK_FETCH_LIMIT } from '../../../app.constant';
+
+@Component({
+    selector: 'app-stepper-form',
```
change here also

ankitdubeyoffice commented on apps/web-giddh/src/app/email-forwarding/components/create/create.component.ts:
```diff
@@ -0,0 +1,494 @@
+        this.bankStatementStore.createUpdateEmailForwardingIsSuccess$.pipe(takeUntil(this.destroyed$)).subscribe((response: unknown) => {
+            if (response && response['uniqueName']) {
+                this.isLoading = false;
+                if (this.isEditMode) {
```
```
this.bankStatementStore.createUpdateEmailForwardingIsSuccess$
    .pipe(takeUntil(this.destroyed$))
    .subscribe((response: { uniqueName?: string } | null | false) => {
        // Always stop loading regardless of response
        this.isLoading = false;
        
        // Early return if no valid response
        if (!response || !response.uniqueName) {
            return;
        }
        
        // Navigate based on mode
        if (this.isEditMode) {
            this.router.navigate(['pages/bank-statement/list']);
        } else {
    const forwardedMail = `${this.emailForwardingForm.value.forwardedMail}${this.forwardedMailDomain}`;
    
    this.router.navigate(['pages/bank-statement/create'], {
        queryParams: {
            forwardedMail,
            response.uniqueName,
            step: 2
        }
    });
        }
    });
```

ankitdubeyoffice commented on apps/web-giddh/src/app/email-forwarding/components/create/create.component.ts:
```diff
@@ -0,0 +1,494 @@
+        });
+    }
+
+    private handleUniqueName(uniqueName: string): void {
```
add doc here

ankitdubeyoffice commented on apps/web-giddh/src/app/bank-statement/components/stepper-form/stepper-form.component.ts:
```diff
@@ -0,0 +1,494 @@
+    public copyEmail(): void {
+        const completeEmail = this.emailForwardingForm.get('forwardedMail')?.value + this.forwardedMailDomain;
+        if (navigator.clipboard) {
+            navigator.clipboard.writeText(completeEmail).then(() => {
```
same for copy apply here also

dvCodeWorld replied:
done

ankitdubeyoffice commented on apps/web-giddh/src/app/email-forwarding/components/create/create.component.ts:
```diff
@@ -0,0 +1,494 @@
+            }, 500);
+            
+        } else {
+            this.toaster.showSnackBar("error", "Verification link not available");
```
use translation here

dvCodeWorld replied:
removoed

ankitdubeyoffice commented on apps/web-giddh/src/app/email-forwarding/components/create/create.component.ts:
```diff
@@ -0,0 +1,494 @@
+    private searchAccount(): void {
+        if (this.accountSearchCalled) {
+            return; // Prevent multiple calls
+        }
```
use also scrolling api call functionality if accounts are many

dvCodeWorld replied:
not required

ankitdubeyoffice commented on apps/web-giddh/src/app/email-forwarding/services/email-forwarding.service.ts:
```diff
@@ -0,0 +1,216 @@
+     * @returns {Observable<BaseResponse<EmailForwardingResponse[], string>>} Observable with list of email forwarding configurations
+     * @memberof EmailForwardingService
+     */
+    public getAllEmailForwarding(): Observable<BaseResponse<EmailForwardingResponse[], string>> {
```
please remove unused doc

dvCodeWorld replied:
done

ankitdubeyoffice commented on apps/web-giddh/src/app/email-forwarding/store/email-forwarding.store.ts:
```diff
@@ -0,0 +1,362 @@
+                            } else {
+                                res?.message && this.toaster.showSnackBar("error", res.message);
+                                return this.patchState({ 
+                                    emailForwardingDetails: undefined, 
```
use null here

dvCodeWorld replied:
correct new any value instead of null

ankitdubeyoffice commented on apps/web-giddh/src/app/email-forwarding/store/email-forwarding.store.ts:
```diff
@@ -0,0 +1,362 @@
+    readonly deleteEmailForwarding = this.effect((data: Observable<string>) => {
+        return data.pipe(
+            switchMap((uniqueName) => {
+                this.patchState({ deleteEmailForwardingIsSuccess: null });
```
use false here if initial is boolean

dvCodeWorld replied:
intentionally

ankitdubeyoffice commented on apps/web-giddh/src/app/theme/form-fields/reactive-dropdown-field/reactive-dropdown-field.component.ts:
```diff
@@ -186,9 +186,6 @@ export class ReactiveDropdownFieldComponent implements ControlValueAccessor, OnI
             }
         });
-        if (filteredOptions.length === 0) {
-            this.writeValue("", false);
```
inform to tester team if you are changing in common component

dvCodeWorld replied:
ok

ankitdubeyoffice commented on apps/web-giddh/src/assets/locale/sidebar-menu/en.json:
```diff
@@ -663,7 +663,30 @@
+            ]
+        },
         {
             "label": "AI OCR for Income",
```
remove old income and expenses also inform to API team for changes in cmd + k

dvCodeWorld replied:
ok

ankitdubeyoffice commented on apps/web-giddh/src/assets/styles/abstracts/_mixins.scss:
```diff
@@ -1045,13 +1045,13 @@
-//  * 
+//  *
 //  * @param {Length} $a - Top-left radius (or all corners if only value)
 //  * @param {Length} $b - Top-right radius (or bottom-left for 2 values)
```
remove commented code

dvCodeWorld replied:
removed

dvCodeWorld committed:
```
refactor: rename bank-statement module to email-forwarding and improve company unique name handling

- Renamed bank-statement module to email-forwarding across routing, file structure, and imports
- Added getCompanyUniqueNameFromUrl helper method to LoginActions, AddCompanyComponent, and AppComponent
- Updated company unique name assignment to prioritize URL query parameter over fallback values
- Added BANK_STATEMENT_HELP_DOC_URL constant for help documentation
- Removed unused SCSS files (data

dvCodeWorld committed:
```
refactor: remove debugger statement from header component login state handler
```

dvCodeWorld committed:
```
refactor: remove debugger statement from company-branch component switchCompany method
```

dvCodeWorld committed:
pr changes
